### PR TITLE
Fix missing argument types in py3 stdlib

### PR DESCRIPTION
**Pull request by JelleZijlstra**
There's a lot more of this in stdlib/2, and I'm still missing a few in _subprocess and decimal.

### CODE REVIEW

File: stdlib/3.4/asyncio/locks.pyi [hunk 1]
```diff
@@ -56,4 +56,4 @@ class Semaphore(_ContextManagerMixin):
     def release(self) -> None: ...
 
 class BoundedSemaphore(Semaphore):
-    def __init__(self, value=1, *, loop: AbstractEventLoop = None) -> None: ...
+    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
```
Comment (gvanrossum) [inline stdlib/3.4/asyncio/locks.pyi]: Should really use `= ...` .

File: stdlib/3.4/asyncio/locks.pyi [hunk 2]
```diff
@@ -49,11 +49,11 @@ class Condition(_ContextManagerMixin):
     def notify_all(self) -> None: ...
 
 class Semaphore(_ContextManagerMixin):
-    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
+    def __init__(self, value: int = ..., *, loop: AbstractEventLoop = ...) -> None: ...
```
Comment (gvanrossum) [inline stdlib/3.4/asyncio/locks.pyi]: But having `= None` implicitly modifies the type to become `Optional[]` so either keep the `None` or use `Optional[AbstractEventLoop]`.

File: stdlib/3.4/asyncio/locks.pyi [hunk 3]
```diff
@@ -49,11 +49,11 @@ class Condition(_ContextManagerMixin):
...
     def locked(self) -> bool: ...
     @coroutine
     def acquire(self) -> Generator[Any, None, bool]: ...
     def release(self) -> None: ...
 
 class BoundedSemaphore(Semaphore):
-    def __init__(self, value: int = 1, *, loop: AbstractEventLoop = None) -> None: ...
+    def __init__(self, value: int = ..., *, loop: AbstractEventLoop = ...) -> None: ...
```
Comment (gvanrossum) [inline stdlib/3.4/asyncio/locks.pyi]: Ditto.

File: stdlib/3.4/asyncio/locks.pyi (hunk 2 above)
Comment (JelleZijlstra) [inline stdlib/3.4/asyncio/locks.pyi]: Oh, I wasn't aware of that. I'll use Optional to be more explicit.

**Comment by ambv** [comment]
Guido, there's 1186 instances of `None` as a default value in typeshed. 1071 are not specifying `Optional[]`. If you want to start enforcing this, we need to add a linter check and codemod everything.

**Comment by gvanrossum** [comment]
Given those stats I don't think we should start enforcing it, but I still think for new code it's better to have the explicit `Optional[]`. Can we do a linter that only flags new code?

**Comment by ambv** [comment]
>  I don't think we should start enforcing it [...] Can we do a linter that only flags new code?

Yes but given our Travis CI setup it's actually easier to codemod everything than to introduce this change. Also, tongue-in-cheek, this sounds like us starting enforcing it ;)

I'd like to codemod type comments to variable annotations anyway some time soon, so I can add this to the list of things TBD.

**Comment by JelleZijlstra** [comment]
I wrote a stublint check to ensure that all defaults are ... but gave up when I saw how many violations there were. If we're OK with codemodding this kind of thing across all of typeshed, I can work on it.

If I do that I'll also ensure the None -> Optional change is handled correctly.

**Pull request merged**